app/assets/javascripts/modules/external_users/claims/FeeCalculator.js

```diff
@@ -132,16 +159,40 @@ moj.Modules.FeeCalculator = {
     });
   },
 
+  buildMiscFeeData: function(data) {
+    data.claim_id = $('#claim-form').data('claimId');
+    var fees = data.fees = [];
+    $('.misc-fee-group:visible').each(function() {
+      fees.push({
+        fee_type_id: $(this).find('select.js-fee-type').val(),
+        quantity: $(this).find('input.js-fee-quantity').val()
+      });
+    });
+  },
+
   // Calculates the "unit price" for a given fixed fee,
   // including fixed fee case uplift fee types.
   calculateUnitPriceFixedFee: function () {
     var self = this;
     var data = {};
     self.addFixedFeeData(data);
 
-    $('.js-fixed-fee-calculator-effectee').each(function() {
+    $('.js-fixed-fee-calculator-effectee').each(function () {
       data.fee_type_id = $(this).closest('.fixed-fee-group').find('select.js-fee-type').val();
       self.unitPriceAjax(data, this);
     });
+  },
+
+  // Calculates the "unit price" for a given fixed fee,
+  // including fixed fee case uplift fee types.
+  calculateUnitPriceMiscFee: function () {
+    var self = this;
+    var data = {};
+    self.buildMiscFeeData(data);
```

Comment:
would make the [addFixedFeeData](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/2491/files#diff-11282b90221b6af098fa25266a35fdd5R178) and buildMiscFeeData functions consistently named. Am easy with either `build..` or `add...`